### ISSUE TITLE
Can't create new service that legacy clients can connect to - problem with binding name?

### ISSUE THREAD

**Issue by jmagaram** [DigDes/SoapCore#178]
I have a .NET 2.0 client that connects to an old .NET Framework 4.6 web site with a WebService.asmx in it. I'm trying to rewrite this web site with ASP.NET Core. I can't get the old client to connect to the new service and I don't know why, even though the new service works just fine when I test it with **wcftestclient**. My guess is the problem has something to do with the binding name of the original site being different than the binding name of the new site. All my attempts to change the binding name in SoapCore doing things like new BasicHttpBinding { Name="WebServiceSoap" } seem to have no effect on the client configuration generated by wcftestclient. Because there must be some important difference between the legacy web site and my new one, I loaded both into wcftestclient and looked at the differences between the client configuration it generated.

Here is the client configuration of my original legacy site:

	<?xml version="1.0" encoding="utf-8"?>
	<configuration>
		<system.serviceModel>
			<bindings>
				<basicHttpBinding>
					<binding name="WebServiceSoap" sendTimeout="00:05:00" />
				</basicHttpBinding>
				<customBinding>
					<binding name="WebServiceSoap12" sendTimeout="00:05:00">
						<textMessageEncoding messageVersion="Soap12" />
						<httpTransport />
					</binding>
				</customBinding>
			</bindings>
			<client>
				<endpoint address="http://www.timesupkidz.com/WebService.asmx"
					binding="basicHttpBinding" bindingConfiguration="WebServiceSoap"
					contract="WebServiceSoap" name="WebServiceSoap" />
				<endpoint address="http://www.timesupkidz.com/WebService.asmx"
					binding="customBinding" bindingConfiguration="WebServiceSoap12"
					contract="WebServiceSoap" name="WebServiceSoap12" />
			</client>
		</system.serviceModel>
	</configuration>

This is the client configuration of the new ASP.NET Core web site:

	<?xml version="1.0" encoding="utf-8"?>
	<configuration>
		<system.serviceModel>
			<bindings>
				<basicHttpBinding>
					<binding name="BasicHttpBinding_WebServiceSoap" sendTimeout="00:05:00" />
				</basicHttpBinding>
			</bindings>
			<client>
				<endpoint address="http://localhost:61453/WebService.asmx" binding="basicHttpBinding"
					bindingConfiguration="BasicHttpBinding_WebServiceSoap" contract="WebServiceSoap"
					name="BasicHttpBinding_WebServiceSoap" />
			</client>
		</system.serviceModel>
	</configuration>

Is the fact that the configuration is different the reason why my legacy client can't connect to the new web site? And if so, how do I make it the same?

When the legacy client tries to connect to the new web site, it gets an exception exception "Client found response content type of 'text/html' but expected 'text/xml' and then buried within the error HTML is InvalidOperationException: No operation found for specified action: http://timesupkidz.com/SubmitErrorReport. Here is my old service code:

	namespace WebApp
	{
		[WebService(Namespace = "http://timesupkidz.com/")]
		[WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
		public class WebService : System.Web.Services.WebService
		{
			[WebMethod]
			public bool SubmitErrorReport(byte[] bytes)
			{
			}
		}
	}

This is my new service code:

	namespace WebApp
	{
		[ServiceContract(Namespace = "http://timesupkidz.com", Name = "WebServiceSoap")]
		public class WebService
		{
			[OperationContract()]
			public bool SubmitErrorReport(byte[] bytes)
			{
			}
		}
	}

And this code appears in my Startup.cs in ConfigureServices:

	services.AddSoapCore();
	services.TryAddSingleton<WebApp.WebService>();

This is in my Configure:

	app.UseSoapEndpoint<WebApp.WebService>("/WebService.asmx", new BasicHttpBinding());

**Comment by jmagaram** [comment]
I've been investigating this more. I uploaded my new ASP.NET Core web site to Azure. When I go into the legacy .NET 2.0 app and choose to update the web service references, my client connects fine. But without doing that I get a bad service request 400. So somehow the new SOAP service isn't configured like the legacy one. I did try this code:

    app.UseSoapEndpoint<WebApp.WebService>("/WebService.asmx", new BasicHttpBinding() { Namespace="Gorilla", Name = "Banana" });

And then when I looked at the published .asmx file on the web site, there is no "Gorilla" or "Banana" in there at all. I can't figure out how to configure the new service to look like the legacy one, even though this should be a pretty simple problem.

**Comment by jmagaram** [comment]
I think I solved this but don't know why it needed to be so difficult. I needed to mark up my service class with some different attributes to get it to be compatible with what I had before. In particular, I needed to add a trailing slash to the Namespace to match the original legacy service. Also, I needed to insert a specific action name for every operation. Otherwise the Action URL ended up being `http://timesupkidz.com/WebService/SubmitErrorReport` rather than the legacy version `http://timesupkidz.com/SubmitErrorReport`. I wish I didn't have to do this for every action. I tried setting the Name attribute on the service to the empty string instead, but that caused an error.

    [ServiceContract(Namespace = "http://timesupkidz.com/")]
    public class WebService
    {
        [OperationContract(Action = "http://timesupkidz.com/SubmitErrorReport")]
        public bool SubmitErrorReport(byte[] bytes)
        {
            return true;
        }
    }
